packages/smooth_app/lib/pages/onboarding/welcome_page.dart

```diff
@@ -0,0 +1,96 @@
+import 'package:smooth_ui_library/util/ui_helpers.dart';
+
+/// Welcome page for first time users.
+class WelcomePage extends StatelessWidget {
+  const WelcomePage();
+
+  @override
+  Widget build(BuildContext context) {
+    final Size screenSize = MediaQuery.of(context).size;
+    // Side padding is 8% of total width.
+    final double sidePadding = screenSize.width * 0.08;
+    // Top padding is 16% of total width.
+    final double topPadding = screenSize.height * 0.016;
+    // Bottom padding is 4% of total width.
+    final double bottomPadding = screenSize.height * 0.04;
+    final AppLocalizations appLocalizations = AppLocalizations.of(context)!;
+    final TextStyle headlineStyle =
+        Theme.of(context).textTheme.headline2!.apply(color: Colors.white);
+    final TextStyle largeButtonTextStyle =
+        Theme.of(context).textTheme.headline3!.apply(color: Colors.white);
+    final TextStyle bodyTextStyle =
+        Theme.of(context).textTheme.bodyText1!.apply(color: Colors.white);
+    return Scaffold(
+      body: Padding(
+        padding: EdgeInsets.only(
+            top: topPadding,
+            bottom: bottomPadding,
+            left: sidePadding,
+            right: sidePadding),
+        child: Column(
+          mainAxisAlignment: MainAxisAlignment.spaceBetween,
+          children: <Widget>[
+            Text(appLocalizations.whatIsOff, style: headlineStyle),
+            Column(
+              crossAxisAlignment: CrossAxisAlignment.start,
+              children: <Widget>[
+                Padding(
+                  padding: const EdgeInsets.only(left: SMALL_SPACE),
+                  child: Text(
+                    appLocalizations.country_label,
+                    style: bodyTextStyle,
+                  ),
+                ),
+                const CountrySelector(),
+                Padding(
+                  padding: const EdgeInsets.only(left: SMALL_SPACE),
+                  child: Text(
+                    appLocalizations.country_selection_explanation,
+                    style: bodyTextStyle,
+                  ),
+                ),
+              ],
+            ),
+            Row(
+              children: <Widget>[
+                Expanded(
+                  child: TextButton(
+                    style: TextButton.styleFrom(
+                      backgroundColor: SmoothTheme.getColor(
+                        Theme.of(context).colorScheme,
+                        SmoothTheme
+                            .MATERIAL_COLORS[SmoothTheme.COLOR_TAG_BLUE]!,
+                        ColorDestination.BUTTON_BACKGROUND,
+                      ),
+                      shape: RoundedRectangleBorder(
+                          borderRadius: BorderRadius.circular(SMALL_SPACE)),
+                      primary: Colors.white,
+                    ),
+                    onPressed: () {
+                      // TODO(jasmeet): Check selected country and navigate to the next onboarding page.
```

Comment:
Actually for the moment you just need to go to the "normal" home page, as storing the country in the preferences would have been done before.

Reply:
Not sure what you mean, the country may not be stored in preferences if the user doesn't select a country and simply clicks `Next`

Reply:
That's precisely the bug I detected:
* I'm in France
* I open the app for the first time
* The welcome page is displayed, and France is displayed as the pre-computed default country
* I don't click on the country; I click on Next, assuming that my "implicit choice" of country=France is stored
* but it probably won't be stored, because I never went to the country list with its `onTap` callback

Reply:
No, see this screenshot. The preselected value was "Please select a country". Although  preselecting (and storing) user's location based country is a  good idea. 

![Screenshot_1640106275](https://user-images.githubusercontent.com/7541730/146971470-52e18210-8d1d-43f7-9d59-9cb184276b13.png)

Reply:
Done